libraries/usersearch/impl/src/main/kotlin/io/element/android/libraries/usersearch/impl/MatrixUserRepository.kt

```diff
@@ -31,24 +32,31 @@ import javax.inject.Inject
 
 @ContributesBinding(SessionScope::class)
 class MatrixUserRepository @Inject constructor(
+    client: MatrixClient,
     private val dataSource: UserListDataSource
 ) : UserRepository {
 
+    private val sessionId = client.sessionId
+
     override suspend fun search(query: String): Flow<List<UserSearchResult>> = flow {
-        // Manually add a fake result with the matrixId, if any
-        val isUserId = MatrixPatterns.isUserId(query)
-        if (isUserId) {
+        // Manually add a fake result with the matrixId, if provided and not the local user
+        val isAnotherUsersId =  query != sessionId.value && MatrixPatterns.isUserId(query)
```

Comment:
Maybe create a MatrixClient.isMe(userId: UserId): Boolean extension method and check for this already existent usage?